src/NeuralNetwork/index.js

```diff
@@ -591,20 +538,13 @@ class DiyNeuralNetwork {
 
     // check to see if layers are passed into the constructor
     // then use those to create your architecture
+    // or use the default layers for the task
     if (!this.neuralNetwork.isLayered) {
-      this.options.layers = this.createNetworkLayers(
-        this.options.layers,
-        this.neuralNetworkData.meta
-      );
-    }
-
-    // if the model does not have any layers defined yet
-    // then use the default structure
-    if (!this.neuralNetwork.isLayered) {
-      this.options.layers = this.addDefaultLayers(
-        this.options.task,
-        this.neuralNetworkData.meta
-      );
+      if (this.options.layers && this.options.layers.length > 2) {
+        this.createNetworkLayers(this.options.layers);
+      } else {
+        this.addDefaultLayers();
+      }
```

Comment:
I removed the modification of `this.options.layers` because it felt weird/sloppy/wrong.  Now I'm slightly concerned that this change could create problems with the `.copy()` method, which uses `this.options` to create the cloned neural network.  Let's put this PR on hold for right now until I have a change to look into it more.

Reply:
Thanks for keeping a close eye on this @lindapaiste! The `copy()` method is critical for the neuroevolution examples I developed for the nature of code book.